File: src/Model/Utils/CustomList.java
package Model.Utils;

import java.util.ArrayList;
import java.util.List;

public class CustomList<T> implements IList<T> {

    private List<T> elems = new ArrayList<>();

    @Override
    public void add(T elem) {
        elems.add(elem);
    }

    @Override
    public Iterable<T> getAll() {
        return elems;
    }

    @Override
    public void remove(T elem) {
        elems.remove(elem);
    }

    @Override
    public String toString() {
        StringBuilder buff = new StringBuilder();

        for (T elem : elems) {
            buff.append(elem).append(" ");
        }
        return buff.toString();
    }
}

File: Toy-Language-C#-Intepreter/Toy-Language-CS-Interpreter/Toy-Language-CS-Interpreter/Model/Containers/CustomList.cs
using System;
using System.Collections;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Toy_Language_CS_Interpreter.Model.Containers
{
    class CustomList<T> : IList<T>
    {
        private List<T> list;

        public CustomList()
        {
            list = new List<T>();
        }

        public void Add(T element)
        {
            list.Add(element);
        }

        public T Get(int index)
        {
            return list.ElementAt(index);
        }

        public IEnumerator<T> GetEnumerator()
        {
            return list.GetEnumerator();
        }

        public void Remove(int index)
        {
            try
            {
                list.RemoveAt(index);
            }
            catch(ArgumentOutOfRangeException e)
            {
                throw new IndexOutOfRangeException("Cannot remove element at that index! Index to big or small");
            }
        }

        public int Size()
        {
            return list.Count;
        }

        IEnumerator IEnumerable.GetEnumerator()
        {
            return list.GetEnumerator();
        }
        public override string ToString()
        {
            StringBuilder builder = new StringBuilder();
            list.ForEach(e => builder.Append(e.ToString()).Append("\r\n"));
            return builder.ToString();
        }
    }
}
